datafusion/sqllogictest/test_files/array.slt

```diff
@@ -1752,6 +1752,101 @@ select array_to_string(make_array(), ',')
 ----
 (empty)
 
+
+## array_union (aliases: `list_union`)
+
+# array_union scalar function #1
+query ?
+select array_union([1, 2, 3, 4], [5, 6, 3, 4]);
+----
+[1, 2, 3, 4, 5, 6]
+
+# array_union scalar function #2
+query ?
+select array_union([1, 2, 3, 4], [5, 6, 7, 8]);
+----
+[1, 2, 3, 4, 5, 6, 7, 8]
+
+# array_union scalar function #3
+query ?
+select array_union([1,2,3], []);
+----
+[1, 2, 3]
+
+# array_union scalar function #4
+query ?
+select array_union([1, 2, 3, 4], [5, 4]);
+----
+[1, 2, 3, 4, 5]
+
+# array_union scalar function #5
+statement ok
+CREATE TABLE arrays_with_repeating_elements_for_union
+AS VALUES
+  ([1], [2]),
+  ([2, 3], [3]),
+  ([3], [3, 4])
+;
+
+query ?
+select array_union(column1, column2) from arrays_with_repeating_elements_for_union;
+----
+[1, 2]
+[2, 3]
+[3, 4]
+
+statement ok
+drop table arrays_with_repeating_elements_for_union;
+
+# array_union scalar function #6
+query ?
+select array_union([], []);
+----
+NULL
+
+# array_union scalar function #7
+query ?
+select array_union([[null]], []);
+----
+[[]]
+
+# array_union scalar function #8
+query ?
+select array_union([null], [null]);
+----
+[]
+
+# array_union scalar function #9
+query ?
+select array_union(null, []);
+----
+NULL
+
+# array_union scalar function #10
+query ?
+select array_union(null, null);
```

Comment:
```
scala> spark.sql("select array_union(array(null), array(null))").show(false)
+-------------------------------------+
|array_union(array(NULL), array(NULL))|
+-------------------------------------+
|[NULL]                               |
+-------------------------------------+
```

Reply:
This is more interesting, because I am not sure null has offsets in the ListArray, I suppose my code just get into the union function, but it exits immediately with empty results, because the offsets are empty...How would you go about it?

Reply:
> ```
> scala> spark.sql("select array_union(array(null), array(null))").show(false)
> +-------------------------------------+
> |array_union(array(NULL), array(NULL))|
> +-------------------------------------+
> |[NULL]                               |
> +-------------------------------------+
> ```

array(null) is not the same as null.

array(null) is like array_union scalar function #8

Reply:
I tried to add an extra branch in my match case like so:

```rust
  (DataType::Null, DataType::Null) => {
            let data = ArrayData::new_empty(&DataType::Null);
            Ok(Arc::new(FixedSizeListArray::from(data)))
        } 
```

but that also produce a NULL response rather than an empty arary, do you have any guidance @comphead ? It's unclear to me how to get an `[]` and also a `[NULL]`